jobs/e2e-test.groovy

```diff
@@ -157,215 +153,82 @@ DEPLOYER_USER = params.DEPLOYER_USERNAME.replace("@", "")
-            sh("ls");
-         }
-      }
-   }
-   return jsonFolders;
-}
-
-def analyzeResults(foldersList) {
-   if (currentBuild.result == 'ABORTED') {
-      // No need to report the results in the case of abort!  They will
-      // likely be more confusing than useful.
-      echo('We were aborted; no need to report results.');
-      return;
-   }
-
-   withTimeout('5m') {
-      // several new files in util and tools
-      kaGit.safePull("webapp/testing/e2e/tools");
-      kaGit.safePull("webapp/testing/e2e/util");
-
-      def notifyResultsArgs = [
-         "./util/notify-e2e-results.ts",
-         "--channel", params.SLACK_CHANNEL,
-         // The URL associated to this Jenkins build.
-         "--build-url", env.BUILD_URL,
-         // The Cypress Cloud build name that will be included at the
-         // beginning of the message.
-         "--label", BUILD_NAME,
-         // The URL we test against.
-         "--url", params.URL,
-         // folders with e2e results
-         "--folders",
-         *foldersList
-      ];
-
-      if (params.DEPLOYER_USERNAME) {
-         // The deployer is the person who triggered the
-         // build (this is only included in the message if
-         // the e2e tests fail).
-         notifyResultsArgs += ["--deployer", params.DEPLOYER_USERNAME];
-      }
-      if (params.SLACK_THREAD) {
-         notifyResultsArgs += ["--thread", params.SLACK_THREAD];
-      }
-
-      // TODO(csilvers): testing/e2e/tools/slack/slack-client.ts
-      // should get the secret directly from gsm, not via an envvar.
-      def slackToken = exec.outputOf([
+   // for security, use only with withGithubToken!
+   GITHUB_TOKEN = exec.outputOf([
+         "gcloud", "--project", "khan-academy",
+         "secrets", "versions", "access", "latest",
+         "--secret", "jenkins_github_webapp_e2e_workflow_runner_token",
+   ]);
+
+   SLACK_TOKEN = exec.outputOf([
           "gcloud", "--project", "khan-academy",
           "secrets", "versions", "access", "latest",
           "--secret", "Slack_api_token_for_slack_owl",
-      ]);
-
-      dir('webapp/testing/e2e') {
-         withEnv(["SLACK_TOKEN=${slackToken}"]) {
-            // notify-e2e-results returns a non-zero rc if it detects
-            // test failures. We set the job to UNSTABLE in that case
-            // (since we don't consider smoketest failures to be
-            // blocking).  But we still keep on running the rest of
-            // this script!
-            catchError(buildResult: "UNSTABLE", stageResult: "UNSTABLE",
-                       message: "There were test failures!") {
-               exec(notifyResultsArgs);
-            }
+   ]);
+}
 
-            // Let notify() know not to send any messages to slack, because we
-            // just did it here.
-            env.SENT_TO_SLACK = '1';
-         }
+def withGitHubToken(body) {
+   wrap([$class: 'MaskPasswordsBuildWrapper', varPasswordPairs: [[password: GITHUB_TOKEN]]]) {
```

Comment:
Nice!

Reply:
Did this require you to add the mask-passwords jenkins plugin?  If so, we should check in that requirement in code -- see Khan/aws-config:jenkins/README.plugins.md

Also, while this masks the password in the console, it doesn't mask it in the actual environment (of course), so someone with access to a jenkins worker machine could see the secret while tihs job is running.  That's not the world's worst security threat, but it's not awesome either.  I try really hard not to pass secrets via envvars. But perhaps it's unavoidable here?